src/api/ledger/parse/utils.js

```diff
@@ -4,6 +4,15 @@ const _ = require('lodash');
 const transactionParser = require('ripple-lib-transactionparser');
 const toTimestamp = require('../../../core/utils').toTimestamp;
 const utils = require('../utils');
+const BigNumber = require('bignumber.js');
+
+function adjustQualityForXRP(quality: string, takerGetsCurrency: string,
+    takerPaysCurrency: string) {
+  const shift = (takerGetsCurrency === 'XRP' ? 6 : 0)
```

Comment:
No, nested ternary is legal, but to make it *logically* correct, it would be:

`const shift =  (takerGetsCurrency === 'XRP') ? (takerPaysCurrency === 'XRP' ? 0 :  6) : 
               (takerPaysCurrency === 'XRP' ? -6 :  0);`

The subtraction method looks more readable to me.